src/button/Button.js

```diff
@@ -72,14 +88,31 @@ export default class Button extends React.Component {
 
     const buttonLogicProps = pick(this.props, ['baseState', 'buttonState', 'onClick', 'stableSuccess', 'timerMillis']);
 
+    const getButtonType = () => {
+      const { type, primary, flat } = this.props;
+      return type || (primary && 'primary') || (flat && 'flat') || 'default';
+    };
+
+    const wrapperStyle = {
+      display: fluid ? 'block' : 'inline-block',
+      width: fluid ? '100%' : null
+    };
+
+    const isIconButton = () => this.props.icon && !this.props.label;
+
+    const className = cx(
```

Comment:
as for the util,
- I add the spread operator in the arguments so it can be used without passing an array, but use the `[].concat(...args)` trick to make it work also with an array
- I replaced the reduce with a simpler map

`export const stateClassUtil = (...classes) => cx([].concat(...classes).map(cl => `is-${cl}`));`